torch/fx/experimental/symbolic_shapes.py

```diff
@@ -3836,6 +3836,10 @@ def _update_var_to_range(self, symbol, vr):
         if upper < 2 and symbol in self.size_like:
             upper = 2
 
+        # If the range is refined to singleton, set replacement
+        if lower == upper:
+            self._set_replacement(symbol, lower, "range_refined_to_singleton")
```

Comment:
This should go after the `else` as it could be that just then the symbol is found to be a constant.